cucumber/authenticators/features/authn_oidc_bad_policy.feature

```diff
@@ -93,3 +93,39 @@ Feature: Users can authenticate with OIDC authenticator
     And I fetch an ID Token
     When I authenticate via OIDC with id token
     Then it is denied
+
+  Scenario: webservice with read and no authenticate permission in policy is denied
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-oidc/keycloak
+      body:
+        - !webservice
+          annotations:
+            description: Authentication service for Keycloak, based on Open ID Connect.
+
+        - !variable
+          id: provider-uri
+
+        - !variable
+          id: id-token-user-property
+
+        - !group users
+
+        - !permit
+          role: !group users
+          privilege: [ read ]
+          resource: !webservice
+
+    - !user alice
+
+    - !grant
+      role: !group conjur/authn-oidc/keycloak/users
+      member: !user alice
+    """
+    And I am the super-user
+    And I successfully set OIDC variables
+    Given I get authorization code for username "alice" and password "alice"
+    And I fetch an ID Token
+    When I authenticate via OIDC with id token
+    Then it is denied
```

Comment:
@orenbm - Thank you!!
what is the meaning of denied? 401? 400? 501?

Reply:
@orenbm this test is in OSS, do you plan to add it to appliance also?

Reply:
No. *if* we'll add tests to the appliance it will be only some vanilla flow

Reply:
> @orenbm - Thank you!!
> what is the meaning of denied? 401? 400? 501?

401 is unauthorized
400 is bad request
501 is not implemented (which we don't throw at all).

Do you think the step should be `Then it is unauthorized`?

Reply:
@orenbm YES!

Reply:
done :)